### ISSUE TITLE
angular 8

### ISSUE THREAD

**Issue by basti1253** [getsaf/shallow-render#137]
Hi there,

first of all thanks for this very nice tool, we use it a lot :)

Any plans about migrating to angular 8 yet?
On a first look this would probably need type updates, drop support for versions < 8 due and updates to the toolchain accordingly.

If you'd need a helping hand I'd dig into it. Maybe this could be a good time to move to jest as well?

regards

Sebastian

**Comment by basti1253** [comment]
ng mocks reference dropping support for older versions: https://github.com/ike18t/ng-mocks/commit/29a5203b7616c7e035a705283f58efef7fe5c25d#diff-d5ca28f100ff8921b33b71d225b2c8ba

**Comment by getsaf** [comment]
Shallow fully supports Angular 8 already (and maintains backward compatibility down to Angular 6).

Also, jest has been supported for a while.

Ng mocks v7 does support ng8 even though it's package.json only mentions ng7.

You should be able to upgrade just fine!

**Comment by getsaf** [comment]
I did notice that it was a little ambiguous in the docs that Angular 8 is supported so I tweaked the version grid at the top.

Also noticed that the front page does not mention Jest. I'll add that soon too.

**Comment by basti1253** [comment]
Pardon. I'm not a native speaker, sorry for the misunderstanding. I've already migrated all my projects to angular 8, I was talking about your shallow-render library.
Along with shallow render npm/yarn installs ng-mocks@7.8. This ng-mocks version lists angular peer dependencies <= version 7: https://github.com/ike18t/ng-mocks/blob/v7.8.0/package.json#L44
 
As a result the package manager warns about unsatisfied peer dependencies. After I checked shallow-renders package.json I realized that it's based on angular 6/karma at the moment.
Simply updating to latest ng-mocks@8.1 would break some unit tests (didn't dig into it yet). The stack trace didn't look really helpfull, though.
That's why I suggested the angular 8 update/jest migration ;)

**Comment by getsaf** [comment]
Yeah, we're in a weird spot right now. Ng-mocks v7 (the one shallow usea) *is compatible with Angular 8* even though the peer dependency warning is shown. Unfortunately, shallow cannot update to ng-mocks 8 without dropping support for Angular 6 and 7.

Once Angular 9 is released, we will drop support for the older versions of Angular and simply keep up with the latest version. This should keep us in lock-step with ng-mocks going forward.

In short, please use the latest version of shallow-render and ignore the peer dependency for Angular 8 projects.

**Comment by basti1253** [comment]
No problem, just wanted to offer a helping hand :) closing for now
